benchmark/csrc/sampler/mapper.cpp

```diff
@@ -0,0 +1,85 @@
+#include <algorithm>
+#include <random>
+#include <type_traits>
+#include <vector>
+
+#include <benchmark/benchmark.h>
+
+#include "pyg_lib/csrc/sampler/cpu/mapper.h"
+
+template <typename node_t, typename scalar_t>
+class BenchmarkMapperCreationAndInsertion : public benchmark::Fixture {
+  static_assert(std::is_integral<node_t>::value &&
+                    std::is_integral<scalar_t>::value,
+                "Integral type required for both node_t and scalar_t");
+
+ protected:
+  void SetUp(const benchmark::State& state) override {
+    static std::random_device rd;
+    static std::mt19937 gen(rd());
+    // fill nodes with whole range two times and shuffle it
+    // this will allow some indices to occur more than once
+    const auto num_nodes = state.range(0);
+    nodes_.resize(num_nodes * 2);
+    std::iota(nodes_.begin(), nodes_.begin() + num_nodes, 0);
+    std::iota(nodes_.begin() + num_nodes, nodes_.end(), 0);
+    std::shuffle(nodes_.begin(), nodes_.end(), gen);
+  }
+
+  void TearDown(const benchmark::State& state) override { nodes_.clear(); }
+
+  void PerformTest(size_t num_nodes, size_t num_entries, size_t samples) {
```

Comment:
I think we can safely remove the `num_entries` argument here. The `num_entries` argument dictates how many samples we approximately want to map. In my understanding, this benchmark assumes that we map all nodes.

Reply:
This benchmark in particular wants to test heuristics for vector usage. We use vector if `num_nodes < 1'000'000` or if `num_entries` is more than 10% of `num_nodes`. By choosing `num_nodes > 1'000'000` and setting explicitly `num_entries` to -1 or `num_nodes` we can test map and vector (respectively) on the same scenarios.

Reply:
I see. That makes sense.